### ISSUE TITLE
Java8 not supported with the provided core component .jar

### ISSUE THREAD

**Issue by cuener** [adobe/aem-core-wcm-components#2892]
**Expected Behaviour**
After Installing core components v. 2.27.0  package bundle will start automatically and resolve all dependencies. 

**Actual Behaviour**
The package installs but the bundle doesn't start, throwing an error about missing a requirement of java 11.

**Steps to Reproduce**
Start an instance of AEM 6.5 running on Java 8
Upload https://codeload.github.com/adobe/aem-core-wcm-components/zip/refs/tags/core.wcm.components.reactor-2.27.0 via packmgr and install.
Note that the bundle doesn't start, and there is an error stack trace in the logs.
Platform and Version
AEM 6.5 on Java 8 SP 20

**Logs taken while reproducing problem**
Cannot start (org.osgi.framework.BundleException: Unable to resolve com.adobe.cq.core.wcm.components.core [850](R 850.0): missing requirement [com.adobe.cq.core.wcm.components.core [850](R 850.0)] osgi.ee; (&(osgi.ee=JavaSE)(version=11)) Unresolved requirements: [[com.adobe.cq.core.wcm.components.core [850](R 850.0)] osgi.ee; (&(osgi.ee=JavaSE)(version=11))])
org.osgi.framework.BundleException: Unable to resolve com.adobe.cq.core.wcm.components.core [850](R 850.0): missing requirement [com.adobe.cq.core.wcm.components.core [850](R 850.0)] osgi.ee; (&(osgi.ee=JavaSE)(version=11)) Unresolved requirements: [[com.adobe.cq.core.wcm.components.core [850](R 850.0)] osgi.ee; (&(osgi.ee=JavaSE)(version=11))]

**[aem-core-wcm-components-core.wcm.components.reactor-2.27.0.zip](https://github.com/user-attachments/files/17661499/aem-core-wcm-components-core.wcm.components.reactor-2.27.0.zip)**
Manifest-Version: 1.0
...
Bundle-ManifestVersion: 2
Bundle-Name: Adobe Experience Manager Core WCM Components Core Bundle
**Bundle-RequiredExecutionEnvironment: JavaSE-11**
Bundle-SymbolicName: com.adobe.cq.core.wcm.components.core
Bundle-Version: 2.27.0
...

**Comment by beo4** [comment]
https://github.com/adobe/aem-core-wcm-components/pull/2755 with this Pull request its introduced. do not understand why this change in conjunction with l18n implies a Java upgrade.
[SITES-21656] OOTB Experience Fragment Localization Issue for /eu/fr Site by @alexandru-stancioiu in https://github.com/adobe/aem-core-wcm-components/pull/2755

https://github.com/adobe/aem-core-wcm-components/releases/tag/core.wcm.components.reactor-2.26.0
the release notes from 2.26.0 do not state the removal of java 8 support

As far as I understand removing of the Java 8 support should be treated as an incompatible change and should be reflected in an upgrade of major version. version 2.26 and 2.27 should be removed as this leads to compications and replaced with 3.0

**Comment by krassib** [comment]
@alexandru-stancioiu - any reason why Java 8 was removed in the bellow commit? My customer is still using Java 8 and cannot upgrade to latest Core Components due to the below change.

https://github.com/adobe/aem-core-wcm-components/commit/d844a0eab1566c3d8bf0cc8ed2973fb6f81e3e63

**Comment by ky940819** [comment]
I suspect it was dropped because of the upgrade because `io.wcm.testing.aem-mock.junit5` was upgraded from `5.1.0` -> `5.5.2`  [(link to change)](https://github.com/adobe/aem-core-wcm-components/pull/2755/files#diff-b5a06276719e759fe07dfe6f75d781be5f83d2215179d82bdb195ad035348214L799-L804).

You can see the compatibility matrix for aem-mock on their website
https://wcm.io/testing/aem-mock/

![image](https://github.com/user-attachments/assets/96811dcb-4782-4938-894d-0679e16d17a1)


Realistically, Java 8 is probably going to have to upgrade eventually - but as it is [still a supported JVM version by AEM](https://experienceleague.adobe.com/en/docs/experience-manager-65/content/implementing/deploying/introduction/technical-requirements) and also still listed as a [supported version for this project](https://github.com/adobe/aem-core-wcm-components/blob/77dd95c70fbf0d4b2a03ea8bb53caca46703dcb1/README.md?plain=1#L96), the issue should probably be corrected.

**Comment by alexandru-stancioiu** [comment]
Hello guys,
[ky940819](https://github.com/ky940819) is right. We needed a fix in that library and then realised newer versions of it were compiled with Java 11, hence we were forced to remove Java 8 support for entire core components project.
AEM 6.5 is fully compatible with JVM Java version 11, so I see no reason not to upgrade.
I missed to update the README, will do it. Thanks for pointing that out, [ky940819](https://github.com/ky940819).

**Comment by krassib** [comment]
@alexandru-stancioiu - this will not bode well with my customer, who is currently stuck on upgrading their Core Components due to the fact that they use JDK8 in all of their servers. They are in the middle of migrating to Core Components and that might be a blocker at some point! Is there a way to make the `io.wcm.testing.aem-mock.junit5` `5.5.2` backward compatible with Java 8 at least for the 2025.

**Comment by vladbailescu** [comment]
@kwin , seems a bit odd to not actually run the tests on the JDK it targets but I guess we could keep Java 8 as compilation target: #2919 (CC @alexandru-stancioiu)

**Comment by kwin** [comment]
@vladbailescu I agree, however in practice I never experienced issues which were not detected due to ITs/UTs being executed with a newer JRE.
